db/migrate/20171124000000_prepare_embed_columns_in_preview_cards.rb

```diff
@@ -0,0 +1,25 @@
+require Rails.root.join('lib', 'mastodon', 'migration_helpers')
+
+# This migration will need servers running a revision equals to or later than
+# commit 067c539ffae4628b7153c6d57f511a92e4693765 for up direction.
+# For down direction, it will need a former revision.
+class PrepareEmbedColumnsInPreviewCards < ActiveRecord::Migration[5.1]
+  include Mastodon::MigrationHelpers
+
+  def up
+    safety_assured do
+      cleanup_concurrent_column_rename :preview_cards, :height, :embed_height
+      cleanup_concurrent_column_rename :preview_cards, :width, :embed_width
+      cleanup_concurrent_column_rename :preview_cards, :html, :embed_html
+    end
+
+    status_ids = Status.joins(:preview_cards)
+                       .where({ preview_cards: { type: :photo } })
+                       .reorder(nil)
+                       .group(:id)
```

Comment:
I think that it is better to use `distinct` instead of `group`.

```
Status.joins(:preview_cards).where({ preview_cards: { type: :photo } }).reorder(nil).distinct.pluck(:id)
```

```
SELECT DISTINCT "statuses"."id" FROM "statuses"
INNER JOIN "preview_cards_statuses" ON "preview_cards_statuses"."status_id" = "statuses"."id"
INNER JOIN "preview_cards" ON "preview_cards"."id" = "preview_cards_statuses"."preview_card_id"
WHERE "preview_cards"."type" = 1
```